services/alerts/src/useAlert.ts

```diff
@@ -1,27 +1,42 @@
-import { useContext } from 'react'
+import { useContext, useRef, useCallback } from 'react'
 import { AlertsManagerContext } from './AlertsManagerContext'
-import { Alert, AlertOptions } from './types'
+import { AlertOptions, AlertsManager, AlertsManagerAlert } from './types'
 
 export const useAlert = (
     message: string | ((props: any) => string),
     options: AlertOptions | ((props: any) => AlertOptions) = {}
 ) => {
-    const alertsManager = useContext(AlertsManagerContext)
+    const alertsManager: AlertsManager = useContext(AlertsManagerContext)
+    const alertRef = useRef(<AlertsManagerAlert>alertsManager.add())
 
-    const show = (props?: any) => {
-        const resolvedMessage = String(
-            typeof message === 'function' ? message(props) : message
-        )
-        const resolvedOptions =
-            typeof options === 'function' ? options(props) : options
+    const show = useCallback(
+        (props?) => {
+            const resolvedMessage = String(
+                typeof message === 'function' ? message(props) : message
+            )
+            const resolvedOptions =
+                typeof options === 'function' ? options(props) : options
 
-        const alert: Alert = {
-            message: resolvedMessage,
-            options: resolvedOptions,
-        }
+            alertsManager.show({
+                ...alertRef.current,
+                message: resolvedMessage,
+                options: resolvedOptions,
+            })
+        },
+        [alertsManager, message, options]
+    )
 
-        alertsManager.add(alert)
-    }
+    const hide = useCallback(() => {
+        alertRef.current.hide(alertRef.current.id)
+    }, [])
+
+    const remove = useCallback(() => {
+        alertRef.current.remove(alertRef.current.id)
+    }, [])
 
-    return { show }
+    return {
+        show,
+        hide,
+        remove,
```

Comment:
I think we should NOT expose `remove` here, it's an implementation detail and I can't think of a valid UX reason to hard-remove an alert without the standard animation